e2e/specs/st_file_uploader.spec.js

```diff
@@ -170,7 +176,12 @@ describe("st.file_uploader", () => {
             force: true,
             subjectType: "drag-n-drop",
             events: ["dragenter", "drop"]
-          })
+          });
+        // Give some time for the first file to upload
+        cy.wait(500);
```

Comment:
Hm, I expect this will mostly work, but two concerns:
1. It could still flake, it's just a lot less likely to based on upload speed?
2. This adds an extra half-second to our CircleCI run always, since our CircleCI tests are currently run sequentially

Could we instead sort the strings as Tim suggested, to resolve these issues?

Reply:
If you expand below, this is supposed to test the order the files are uploaded. I'm uploading them in a specific order and then I'm verifying the order below so it is manually sorted

Reply:
Is it important to test the order that files are uploaded? I'm also skeptical of using timers to mitigate non-deterministic tests. 

A possibly more-correct way to test this would be to not upload the second file until the server indicates it's received the first file - possibly by changing some text on the page or something. Or maybe we need to add "uploaded at" timestamp values to `UploadedFileRec`, so that the test can sort. But I wonder if this something we actually care about testing; I don't think make any claims in the API about the ordering of multiple files, and I'm not sure what the use case would be?

Reply:
We don't make any claims about the sorting order so we don't have to test this but I do believe this is desirable behavior. Preserving this sorting allows streamlit developers to provide a better UX to their users. For example if someone uploads 2 files and each file results in one table being printed, when a 3rd file is uploaded, the table for the new file should be printed after the first 2. In order to do this, we would have to preserve the order of the list so that the python script would process the files in the same order on rerun.

We could add an uploaded at timestamp but sounds like that would be solely for the purposes of this test case? I think the waiting to upload the second file until we have received the first is the correct way to go instead and will update.